src/flask_pyoidc/provider_configuration.py

```diff
@@ -164,22 +187,26 @@ def ensure_provider_metadata(self):
     def registered_client_metadata(self):
         return self._client_metadata
 
-    def register_client(self, redirect_uris, extra_parameters=None):
+    def register_client(self, client: Client):
+
         if not self._client_metadata:
-            if 'registration_endpoint' not in self._provider_metadata:
+            if not self._provider_metadata['registration_endpoint']:
                 raise ValueError("Can't use dynamic client registration, provider metadata is missing "
                                  "'registration_endpoint'.")
 
             registration_request = self._client_registration_info.to_dict()
-            registration_request['redirect_uris'] = redirect_uris
-            if extra_parameters:
-                registration_request.update(extra_parameters)
 
-            resp = self.requests_session \
-                .post(self._provider_metadata['registration_endpoint'],
-                      json=registration_request,
-                      timeout=self.DEFAULT_REQUEST_TIMEOUT)
-            self._client_metadata = ClientMetadata(redirect_uris=redirect_uris, **resp.json())
+            # Send request to register the client dynamically.
+            registration_response = client.register(
+                url=self._provider_metadata['registration_endpoint'],
+                **registration_request)
+            logger.debug(registration_response.to_dict())
+            # post_logout_redirect_uris is not returned to the request so
+            # explicitly provide this as a keyword argument to initialise
+            # ClientMetadata.
+            self._client_metadata = ClientMetadata(
+                **registration_response.to_dict(),
+                post_logout_redirect_uris=registration_request['post_logout_redirect_uris'])
```

Comment:
This one as well now: `post_logout_redirect_uris=registration_request['post_logout_redirect_uris']` should not be needed.

Reply:
Some IdPs may not return empty list if the user doesn't provide post_logout_redirect_uris  in `ClientRegistrationInfo` which will cause `ClientMetadata` to fail.

Here's the new code to ensure that `ClientMetada` takes `post_logout_redirect_uris` from the code instead of json response:

```python
# Send request to register the client dynamically.
registration_response = client.register(
    url=self._provider_metadata['registration_endpoint'],
    **registration_request)
logger.debug(registration_response.to_dict())
# Some IdPs may not return post_logout_redirect_uris to the
# response so explicitly patch the response to take
# post_logout_redirect_uris from the code. Even if it exists in
# the response, it will be replaced.
registration_response['post_logout_redirect_uris'] = registration_request[
    'post_logout_redirect_uris']
self._client_metadata = ClientMetadata(
    **registration_response.to_dict())
```

What do you think?

Reply:
If the IdP doesn't return `post_logout_redirect_uris` or returns a subset of the URIs sent in the request it would be incorrect to overwrite it like this. It could indicate that provider either doesn't support `post_logout_redirect_uris` (and hence doesn't register any) or that they apply some policy to which URIs may be registered for a client (only accepting a subset of the ones that were sent).
Hence, the response from the provider should be taken as-is since we on this side can't be sure it wasn't intentional from the provider to not include the `post_logout_redirect_uris` in the response.

Reply:
Done 🙂